src/sentry/snuba/discover.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import absolute_import
+
+import six
+
+from copy import deepcopy
+from sentry.api.event_search import get_filter, resolve_field_list
+from sentry.utils.snuba import (
+    Dataset,
+    DISCOVER_COLUMN_MAP,
+    QUOTED_LITERAL_RE,
+    get_function_index,
+    raw_query,
+)
+
+ALLOWED_INTERNALS = set(["project_id", "event_id"])
+
+
+def resolve_column(col):
+    """
+    Resolve a public schema name to the discover dataset.
+    unknown columns are converted into tags expressions.
+    """
+    if not col:
+        return ""
+    if col.startswith("tags[") or col in ALLOWED_INTERNALS or QUOTED_LITERAL_RE.match(col):
+        return col
+    return DISCOVER_COLUMN_MAP.get(col, u"tags[{}]".format(col))
```

Comment:
Are the columns already converted to `"tags[...]"` syntax by the time we get here or not? Line 25 implies that that's already the case

Reply:
line 25 is aiming to handle users using the `tags[foo]:value` syntax in their query condition string. There isn't an integration test for this format, I'll add one.